src/components/LmsConfigurations/index.test.jsx

```diff
@@ -83,21 +113,48 @@ describe('<LmsConfigurations /> ', () => {
     expect(wrapper.find('BlackboardIntegrationConfigForm').props().config).toEqual(wrapper.state().blackboardConfig);
   });
 
+  it('get the Canvas LMS configuration when present', async () => {
+    LmsApiService.fetchMoodleConfig.mockRejectedValue(notFoundMoodleResponse);
+    LmsApiService.fetchBlackboardConfig.mockRejectedValue(notFoundBlackboardResponse);
+    LmsApiService.fetchCanvasConfig.mockResolvedValue(canvasResponse);
+    LmsApiService.fetchSuccessFactorsConfig.mockRejectedValue(notFoundSuccessFactorsResponse);
+
+    const wrapper = mount(<LmsConfigurations enterpriseId="testEnterpriseId" />);
+    await waitForAsync();
+    wrapper.update();
+    expect(wrapper.state().canvasConfig).toEqual(canvasResponse.data.results[0]);
+    expect(wrapper.find('CanvasIntegrationConfigForm').props().config).toEqual(wrapper.state().canvasConfig);
+  });
+
+  it('get the Success Factors LMS configuration when present', async () => {
+    LmsApiService.fetchMoodleConfig.mockRejectedValue(notFoundMoodleResponse);
+    LmsApiService.fetchBlackboardConfig.mockRejectedValue(notFoundBlackboardResponse);
+    LmsApiService.fetchCanvasConfig.mockRejectedValue(notFoundCanvasResponse);
+    LmsApiService.fetchSuccessFactorsConfig.mockResolvedValue(successFactorsResponse);
+
+    const wrapper = mount(<LmsConfigurations enterpriseId="testEnterpriseId" />);
+    await waitForAsync();
+    wrapper.update();
+    expect(wrapper.state().sapsfConfig).toEqual(successFactorsResponse.data.results[0]);
+    expect(wrapper.find('SuccessFactorsIntegrationConfigForm').props().config).toEqual(wrapper.state().sapsfConfig);
+  });
+
   it('return Error Page when the Moodle LMS fetch request fails with error (not 404)', async () => {
     LmsApiService.fetchMoodleConfig.mockRejectedValue(badMoodleResponse);
 
     // We want to ensure that the Moodle fetch config method will cause the error page to display
     // if it encounters an unexpected error while the other config fetch methods succeed.
     LmsApiService.fetchBlackboardConfig.mockResolvedValue(blackboardResponse);
-    LmsApiService.fetchCanvasConfig.mockResolvedValue(canvasResponse);
+    LmsApiService.fetchCanvasConfig.mockRejectedValue(notFoundCanvasResponse);
+    LmsApiService.fetchSuccessFactorsConfig.mockRejectedValue(notFoundSuccessFactorsResponse);
 
     const wrapper = mount(<LmsConfigurations enterpriseId="testEnterpriseId" />);
     await waitForAsync();
     wrapper.update();
 
-    // Expect the moodle config to be false, while the others to be true
     expect(wrapper.state().moodleConfig).toBeFalsy();
     expect(wrapper.find('ErrorPage').length).toBe(1);
+    expect(wrapper.find('MoodleIntegrationConfigForm').length).toBe(0);
```

Comment:
What are we expecting the canvas and SuccessFactors forms to look like here? Are they supposed to be present, or is it just an error page? Can we add expects for that?

Reply:
(Perhaps that should be a different test/tests)

Reply:
We're not expecting them that's why the length is 0. I was playing around with the logic here to get this expect logic to work in the intended way which is: "I should not see a form here. I should see an error page and only an error page"

Reply:
I think I struggled getting any sort of "exists" logic instead of this length usage.

Reply:
This appears to only test that the length of the Moodle form is zero. Could we add some expects to test that the Canvas and SuccessFactors forms are also not present, despite positive API results?